### PR TITLE
hibernate-resume: add & use devnode_same to compare device nodes

### PR DESCRIPTION
**Pull request by YHNdnzj**
Addresses https://github.com/systemd/systemd/pull/28321#discussion_r1257866179

Fixes https://github.com/systemd/systemd/issues/28340

### CODE REVIEW

**Comment by YHNdnzj** [comment]
(I didn't call `path_equal` since we're going to `stat` anyway. It should be more efficient to compare `st_rdev` directly.)

**Comment by poettering** [comment]
I'd really do a path_equal() check here too, because device nodes tend to appear at some point in time only, they are generally not available always due to plug/play, linux device model, devtmpfs and so on. If we check by path_equal() we have the advantage that as long as people specify two identical paths they can avoid any timing/race issues and immediately get the expected answer.

**Comment by YHNdnzj** [comment]
> I'd really do a path_equal() check here too, because device nodes tend to appear at some point in time only, they are generally not available always due to plug/play, linux device model, devtmpfs and so on. If we check by path_equal() we have the advantage that as long as people specify two identical paths they can avoid any timing/race issues and immediately get the expected answer.

Hmm, makes sense. Updated.

**Review by yuwata — APPROVED** [review]

File: src/hibernate-resume/hibernate-resume-generator.c [hunk 1]
```diff
@@ -170,7 +171,14 @@ static int parse_efi_hibernate_location(void) {
                 arg_resume_device = TAKE_PTR(device);
                 arg_resume_offset = location.offset;
         } else {
-                if (!path_equal_or_inode_same(arg_resume_device, device, 0))
+                r = path_equal(arg_resume_device, device);
```
Comment (poettering) [inline src/hibernate-resume/hibernate-resume-generator.c]: path_equal() returns a boolean, never an error

**Pull request merged**